woodwork/table_accessor.py

```diff
@@ -153,16 +153,30 @@ def set_index(self, new_index):
         and clears the tag from any previously set index column.
         Setting a column as the index column will also cause any previously set standard
         tags for the column to be removed.
+        Clears the DataFrame's index by passing in None.
 
         Args:
-            index (str): The name of the column to set as the index
+            new_index (str): The name of the column to set as the index
         """
         self._schema.set_index(new_index)
 
         if self._schema.index is not None:
             _check_index(self._dataframe, self._schema.index)
         self._set_underlying_index()
 
+    def set_time_index(self, new_time_index):
+        '''Sets the time index column of the DataFrame. Adds the 'time index' semantic tag to the column
+        and clears the tag from any previously set time index column.
+        Clears the DataFrame's time index by passing in None.
+
+        Args:
+            new_time_index (str): The name of the column to set as the time index
+        '''
+        self._schema.set_time_index(new_time_index)
+
+        if self._schema.time_index is not None:
+            _check_time_index(self._dataframe, self._schema.time_index)
```

Comment:
I'm not 100% convinced that there needs to be a method on the Accessor. The `_check_time_index` in `schema.py` which gets called first, checks whether the column is a Datetime logical type or has the `numeric` standard tag, which I think might be good enough in this case, since we'll have previously confirmed that the logical type being used is valid.

Our checks in the accessor's `_check_time_index` basically make sure that the potential time index column is a valid Datetime or numeric.

Reply:
Makes sense. If we can eliminate this, I'd say let's do it.

Reply:
Yep, I haven't thought of a reason to keep it. Removed!